src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -112,6 +139,83 @@ public BlobFeedAction(string expectedFeedUrl, string accountKey, MSBuild.TaskLog
             return !Log.HasLoggedErrors;
         }
 
+        public async Task UploadAssets(ITaskItem item, SemaphoreSlim clientThrottle, bool allowOverwrite = false)
+        {
+            string relativeBlobPath = item.GetMetadata("RelativeBlobPath");
+            if (string.IsNullOrEmpty(relativeBlobPath))
+                throw new Exception($"Metadata 'RelativeBlobPath' is missing for item '{item.ItemSpec}'.") ;
+            relativeBlobPath = $"{feed.RelativePath}{relativeBlobPath.Replace("\\", "/")}"; 
+            
+            Log.LogMessage($"Uploading {relativeBlobPath}");
+
+            await clientThrottle.WaitAsync();
+
+            // this defines the lease for 15 seconds (max is 60) and 3000 milliseconds between requests
+            CloudTestTasks.AzureBlobLease blobLease = new CloudTestTasks.AzureBlobLease(
```

Comment:
Is this pulling in a new binary dependency or is this source sharing?

Reply:
Nevermind I see it is source you pulled in.

Reply:
Shared code. Namespace is different so I had to add it but is not a binary dependency